server/images.go

```diff
@@ -141,11 +149,19 @@ func (m *Model) CheckCapabilities(want ...model.Capability) error {
 		}
 	}
 
+	var err error
 	if len(errs) > 0 {
-		return fmt.Errorf("%w %w", errCapabilities, errors.Join(errs...))
+		err = fmt.Errorf("%w %w", errCapabilities, errors.Join(errs...))
 	}
 
-	return nil
+	if slices.Contains(errs, errCapabilityThinking) {
+		if m.Config.ModelFamily == "qwen3" || model.ParseName(m.Name).Model == "deepseek-r1" {
+			// append a message to the existing error
+			return fmt.Errorf("%w. Pull the model again to get the latest version with full thinking support", err)
```

Comment:
```suggestion
			return fmt.Errorf("%w. You may need to pull the model again to get the latest version with support for thinking", err)
```

minor suggestion in case the model doesn't support thinking at all

Reply:
note that this path is the error message only for qwen3/deepseek-r1. Were you thinking about this tweak for all models or were you thinking about the case where perhaps the user is using a custom model that was based on an old qwen3 or something?

Reply:
Shoot, I did not realize that. It looks great as is then!